servicetalk-data-protobuf-jersey/src/main/java/io/servicetalk/data/protobuf/jersey/ProtobufSerializerMessageBodyReaderWriter.java

```diff
@@ -0,0 +1,240 @@
+                providers.getContextResolver(ProtobufSerializerFactory.class, mediaType);
+
+        return contextResolver != null ? contextResolver.getContext(ProtobufSerializerFactory.class) :
+                ProtobufSerializerFactory.PROTOBUF;
+    }
+
+    private static Publisher<Buffer> toBufferPublisher(final InputStream is, final BufferAllocator a) {
+        return fromInputStream(is).map(a::wrap);
+    }
+
+    private static <T> Single<T> deserialize(
+            final Publisher<Buffer> bufferPublisher, final Deserializer<T> deserializer, final int contentLength,
+            final BufferAllocator allocator) {
+        return bufferPublisher
+                .collect(() -> newBufferForRequestContent(contentLength, allocator), Buffer::writeBytes)
+                .map(buf -> {
+                    try {
+                        return deserializer.deserialize(buf, allocator);
+                    } catch (final NoSuchElementException e) {
+                        throw new BadRequestException("No deserializable Protobuf content", e);
+                    } catch (final SerializationException e) {
+                        // SerializationExceptionMapper can't always tell for sure that the exception was thrown because
+                        // of bad user data: here we are deserializing user data so we can assume we fail because of it
+                        // and immediately throw the properly mapped JAX-RS exception
+                        throw new BadRequestException("Invalid Protobuf data", e);
+                    }
+                });
+    }
+
+    static Buffer newBufferForRequestContent(final int contentLength,
+                                             final BufferAllocator allocator) {
+        return contentLength == -1 ? allocator.newBuffer() : allocator.newBuffer(contentLength);
+    }
+
+    // visible for testing
+    static <T> T deserializeObject(final Publisher<Buffer> bufferPublisher, final Deserializer<T> deserializer,
+                                   final int contentLength, final BufferAllocator allocator) {
+        return awaitResult(deserialize(bufferPublisher, deserializer, contentLength, allocator).toFuture());
+    }
+
+    private static boolean isSse(ContainerRequestContext requestCtx) {
+        final ResourceMethod method = ((ExtendedUriInfo) requestCtx.getUriInfo()).getMatchedResourceMethod();
+        return method != null && method.isSse();
+    }
+
+    private static boolean isSupportedMediaType(final Class<?> type, final MediaType mediaType) {
+        return (!Publisher.class.equals(type) &&
+                (mediaType.getType().equalsIgnoreCase(APPLICATION_PROTOBUF_TYPE.getType()) &&
+                    (mediaType.getSubtype().equalsIgnoreCase(APPLICATION_PROTOBUF_TYPE.getSubtype()) ||
+                    mediaType.getSubtype().toLowerCase().endsWith('+' + APPLICATION_PROTOBUF_TYPE.getSubtype())))) ||
+                (Publisher.class.equals(type) &&
```

Comment:
I can see users who need to use delimited format for a single object. Publisher is not required for that.

Reply:
The encoding for var-int is different than the no-delimiter case (e.g. single object) and is not interchangeable from a serialization perspective. If you know you always have a single object you wouldn't need delimited encoding. The var-int delimited encoding is only implemented for the streaming serializers and provides the framing to send multiple objects over a single stream, if a single object is needed with var-int decoding `Publisher.from(pojo)` is the supported way.